### ISSUE TITLE
Any chance you can create a Docker image that runs on a Raspberry Pi??

### ISSUE THREAD

**Issue by bigdant** [JasonHHouse/gaps#127]
Hi,

Any chance you can create a docker image that runs on a Raspberry pi ?

armhf runs on portainer on a Rapsberry pi 3.

Architecture | Tag
x86-64 | amd64-latest
arm64 | arm64v8-latest
armhf | arm32v7-latest

Cheers

**Comment by JasonHHouse** [comment]
Let me look for a container. The trick is that some JVMs aren't as memory-efficient and I didn't want to use ARM but bloat your memory usage.

**Comment by bigdant** [comment]
Managed to get Radarr, Sonarr, Lidarr, Deluge, etc all running on Portainer on a Pi 3.

All using Linuxserver dockers (https://hub.docker.com/u/linuxserver)

Would be great to get Gaps working on it too :)

Cheers

D.

**Comment by JasonHHouse** [comment]
Thanks, that's helpful. I can check out their docker bases and look at that as an option. Assuming it has a JVM.

**Comment by JasonHHouse** [comment]
@bigdant Can you give this a try `docker pull housewrecker/gaps:arm-latest`? This has linux/arm64 and linux/arm/v7. If it works, I'll add it to the normal build process for gaps.

**Comment by bigdant** [comment]
@JasonHHouse - still not working.

Getting a (standard_init_linux.go:211: exec user process caused "exec format error") error message

**Comment by JasonHHouse** [comment]
@bigdant Can you re-pull and try? `docker pull housewrecker/gaps:arm-latest` 

TL;DR
If this doesn't work, I may be out of luck for a bit. I don't have ARM hardware and might need to get some.

Longer Reasoning
I'm running on a MacBook Pro and Ryzen system. Docker lets me use an experimental builder to make multiple arch's like ARM. However, I'm still limited by which arch's are supported in docker and with my system. I may need to pick up something like a Rasp Pi or make a dev VM in AWS to get more arch's support for building. With ARM, I don't have a way to test either to confirm what's working for you. So, I'll keep you posted and I'm working on it, just may take a bit to put some money towards a solution.

**Comment by bigdant** [comment]
That seems to have fixed it !!! I'm in.

The app is awesome, does it have the ability to update Plex with new collections from within the app ??

Does it just produce the list or can it be integrated with Sonarr / Radarr, etc ?

Cheers

D.

**Comment by JasonHHouse** [comment]
Awesome, I'll push that docker version from now on. 

Right now, it just makes the list for Radarr to use, because I only have movie support right now. Sonarr is on my list to add. 

I'd like to add Plex direct integration as well, from what I've seen though, the API is limited. I can put that on the list to check out though.

**Comment by Floppy** [comment]
I don't like to reopen old threads, but I've just tried using `housewrecker/gaps:arm-latest` on my pi 4 running buster, and it's giving me an `exec format error` as reported here. Should this still be working?

**Comment by JasonHHouse** [comment]
It should be. I don't have an arm device so I'll need to setup an emulator. I'll check it out though.

**Comment by JasonHHouse** [comment]
@Floppy I tested on my arm Chromebook without issue. Can you send some logs or more details on what you are seeing?

**Comment by Floppy** [comment]
OK - I've manually done the following, copy/pasting the README instructions, on a machine that had no previous pull:

```
root@wonko:~# docker pull housewrecker/gaps:arm-latest
arm-latest: Pulling from housewrecker/gaps
2d6a538ce9cb: Pull complete 
11d2d1780e64: Pull complete 
54e91fa788b8: Pull complete 
4553441b6f08: Pull complete 
518a3650bcb5: Pull complete 
3d27188855a7: Pull complete 
ed269585b52f: Pull complete 
622192ecc9cc: Pull complete 
938e2d1e87e0: Pull complete 
ac3e1bedf426: Pull complete 
385a1dbc6ade: Pull complete 
008b8f4034ab: Pull complete 
Digest: sha256:ba10d2d074bc7ce2ea670a1049b4881e99d874143fb7c05a06e82f33ba628f92/usr/data housewrecker/gaps:latest
Status: Downloaded newer image for housewrecker/gaps:arm-latest
docker.io/housewrecker/gaps:arm-latest
root@wonko:~# docker run -d -p 8484:8484 --name mygaps --expose 32400 -v /root/gaps:/usr/data housewrecker/gaps:arm-latest
f7f68e15844a1c1d9f48bb310e4b40ed998c268fed8de85ea6532be2b460dbc4
root@wonko:~# docker logs mygaps
standard_init_linux.go:211: exec user process caused "exec format error"
```

This is a pi 3B+, running raspbian buster. Docker version: `19.03.4, build 9013bf5`, arch: `armv7l`

**Comment by JasonHHouse** [comment]
That looks like an issue of the container given the arch. I made another build can you try this one.

`docker pull housewrecker/gaps:armhf-latest`

**Comment by Floppy** [comment]
That worked! Thanks :)

**Comment by JasonHHouse** [comment]
Awesome, I'll make sure to add those builds to all future releases.
